### PR TITLE
Using `Option` on FFI

### PR DESCRIPTION
**Pull request by robertbastian**
This is ABI compatible but API breaking in C++ and arguably JS.

### CODE REVIEW

**Review by Manishearth — CHANGES_REQUESTED** [review]
we can't land this until 2.0

**Comment by robertbastian** [comment]
Do you consider this breaking in JS?

I'm happy to change the CPP1 generator to translate `Option` to `diplomat::Result<T, std::monostate>` to not have a breaking change there.

**Comment by Manishearth** [comment]
I'm personally ok with breaking JS. I'll leave it up to @sffc

File: ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts [hunk 1]
```diff
@@ -27,10 +26,9 @@ export class ICU4XScriptExtensionsSet {
...
   /**
 
-   * Get script at index, returning an error if out of bounds
+   * Get script at index
 
    * See the {@link https://docs.rs/icu/latest/icu/properties/script/struct.ScriptExtensionsSet.html#method.iter Rust documentation for `iter`} for more information.
-   * @throws {@link FFIError}<void>
    */
-  script_at(index: usize): u16 | never;
+  script_at(index: usize): u16 | undefined;
```
Comment (sffc) [inline ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts:33]: Issue: the implementation returns null not undefined

**Review by sffc — APPROVED** [review]
The JS change is breaking but the TS should help and I think we don't have very many JS users and the impacted functions are sufficiently in the corner that I'm okay with a changelog notice

File: ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts (hunk 1 above)
Comment (robertbastian) [reply ffi/capi/bindings/js/ICU4XScriptExtensionsSet.d.ts:33]: ah that's a bug in Diplomat then

**Review by Manishearth — APPROVED** [review]

**Pull request merged**